GeeksCoreLibrary/Modules/Templates/Services/TemplatesService.cs

```diff
@@ -1437,6 +1470,85 @@ public async Task<List<Template>> GetTemplateUrlsAsync()
         }
 
+        /// <inheritdoc />
+        public async Task<bool> ComponentRenderingShouldBeLoggedAsync(int componentId)
+        {
+            var logRenderingOfComponentsSetting = await objectsService.FindSystemObjectByDomainNameAsync($"log_rendering_of_components_{gclSettings.Environment}");
+            if (String.IsNullOrWhiteSpace(logRenderingOfComponentsSetting))
+            {
+                logRenderingOfComponentsSetting = await objectsService.FindSystemObjectByDomainNameAsync("log_rendering_of_components");
+                if (String.IsNullOrWhiteSpace(logRenderingOfComponentsSetting))
+                {
+                    return false;
+                }
+            }
+            
+            if (String.Equals("all", logRenderingOfComponentsSetting, StringComparison.OrdinalIgnoreCase) || String.Equals("true", logRenderingOfComponentsSetting, StringComparison.OrdinalIgnoreCase))
+            {
+                return true;
+            }
+
+            var listOfComponentIdsToLog = logRenderingOfComponentsSetting.Split(",").Select(value => !Int32.TryParse(value, out var id) ? 0 : id);
+            return listOfComponentIdsToLog.Contains(componentId);
+        }
+
+        /// <inheritdoc />
+        public async Task<bool> TemplateRenderingShouldBeLoggedAsync(int templateId)
+        {
+            var logRenderingOfTemplatesSetting = await objectsService.FindSystemObjectByDomainNameAsync($"log_rendering_of_templates_{gclSettings.Environment}");
+            if (String.IsNullOrWhiteSpace(logRenderingOfTemplatesSetting))
+            {
+                logRenderingOfTemplatesSetting = await objectsService.FindSystemObjectByDomainNameAsync("log_rendering_of_templates");
+                if (String.IsNullOrWhiteSpace(logRenderingOfTemplatesSetting))
+                {
+                    return false;
+                }
+            }
+            
+            if (String.Equals("all", logRenderingOfTemplatesSetting, StringComparison.OrdinalIgnoreCase) || String.Equals("true", logRenderingOfTemplatesSetting, StringComparison.OrdinalIgnoreCase))
+            {
+                return true;
+            }
+
+            var listOfTemplateIdsToLog = logRenderingOfTemplatesSetting.Split(",").Select(value => !Int32.TryParse(value, out var id) ? 0 : id);
+            return listOfTemplateIdsToLog.Contains(templateId);
+        }
+
+        /// <inheritdoc />
+        public async Task AddTemplateOrComponentRenderingLogAsync(int componentId, int templateId, int version, DateTime startTime, DateTime endTime, long timeTaken, string error = "")
+        {
+            try
+            {
+                var userData = await accountsService.GetUserDataFromCookieAsync();
+                
+                var tableName = componentId > 0 ? WiserTableNames.WiserDynamicContentRenderLog : WiserTableNames.WiserTemplateRenderLog;
+                await databaseHelpersService.CheckAndUpdateTablesAsync(new List<string> {tableName});
```

Comment:
Is het niet beter om dit te controleren bij het opstarten van de website? Nu wordt dit meerdere keren gedaan bij het openen van een pagina (template + voor alle dynamische componenten in die template). De impact zal klein zijn maar naarmate er meer dynamische componenten binnen een template zitten kan dit wel oplopen.